static/js/server_events_dispatch.js

```diff
@@ -317,6 +318,13 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
                 message_list.narrowed.rerender();
             }
         }
+        if (event.setting_name === 'emoji_conversion') {
+            // Rerender the whole message list UI
+            home_msg_list.rerender();
+            if (current_msg_list === message_list.narrowed) {
+                message_list.narrowed.rerender();
+            }
+        }
```

Comment:
This doesn't changes exist messages I think? Earlier messages don't have the converted emoji name in the source, so they won't display the emoji.